### ISSUE TITLE
Check CAN Filters, Fix if necessary

### ISSUE THREAD

**Issue by stieg** [autosportlabs/RaceCapture-Pro_firmware#737]
According to https://forum.autosportlabs.com/viewtopic.php?p=24320 the filters on CAN are not functioning correctly.  Need to verify them and fix them if something is indeed broken.

**Comment by stieg** [comment]
Built a loopback device to send messages from CAN 1 to CAN 0 on my MK2.  Wrote the following script to test out the filtering:

```lua
--- SETTINGS ---
-- How frequently we poll for CAN messages
tickRate = 1

-- The CAN baud rate
CAN0_baud = 500000
CAN1_baud = 500000

--CAN channel to listen on. 0=first CAN channel, 1=second
CAN_0 = 0
CAN_1 = 1

-- CAN Filter and Addressess
CAN_REG_ADDR_GOOD_1 = 0xC0
CAN_REG_ADDR_GOOD_2 = 0xCF
CAN_REG_ADDR_FAIL_1 = 0xF0
CAN_REG_FILTER_MASK = 0xF0
CAN_REG_FILTER_VAL  = 0XC0
CAN_REG_FILTER_ID   = 1

CAN_EXT_ADDR_GOOD_1 = 0xC0000
CAN_EXT_ADDR_GOOD_2 = 0xCFFFF
CAN_EXT_ADDR_FAIL_1 = 0xF0000

CAN_EXT_FILTER_MASK = 0xF0000
CAN_EXT_FILTER_VAL  = 0XC0000
CAN_EXT_FILTER_ID   = 2

---- CODE ----
initCAN(CAN_0, CAN0_baud)
initCAN(CAN_1, CAN1_baud)
setCANfilter(CAN_0, CAN_REG_FILTER_ID, 0, CAN_REG_FILTER_VAL, CAN_REG_FILTER_MASK)
setCANfilter(CAN_0, CAN_EXT_FILTER_ID, 1, CAN_EXT_FILTER_VAL, CAN_EXT_FILTER_MASK)
setTickRate(tickRate)

function print_can_msg(chan, id, data)
    print(getUptime())
    print(' [CAN ')
    print(chan)
    print(']: ')
    print(id .. ' => [')

    local comma = false
    for i = 1,#data do
        print(comma and ', ' or '')
        comma = true
        print(data[i])
    end

    print(']\n')
end

function rx_can(chan, wait)
    	local id, ext, data = rxCAN(chan, wait)
	if not id then return false end
        print_can_msg(chan, id, data)
    	return true
end

function tx_can_msgs(val)
	--txCAN(channel, id, isExtended, data, [timeout] )
	data_arr = {val}
	txCAN(CAN_1, CAN_REG_ADDR_GOOD_1, 0, data_arr)
	txCAN(CAN_1, CAN_REG_ADDR_GOOD_2, 0, data_arr)
	txCAN(CAN_1, CAN_REG_ADDR_FAIL_1, 0, data_arr)

	txCAN(CAN_1, CAN_EXT_ADDR_GOOD_1, 1, data_arr)
	txCAN(CAN_1, CAN_EXT_ADDR_GOOD_2, 1, data_arr)
	txCAN(CAN_1, CAN_EXT_ADDR_FAIL_1, 1, data_arr)
end

function onTick()
	value = getUptime() % 256
	println("Uptime value is: " .. value)
    	tx_can_msgs(value)
    	while rx_can(0, 0) do end
end
```

And got the following output:

```txt
Uptime value is: 152.0
13765791.0 [CAN 0.0]: 192.0 => [152.0]
13765794.0 [CAN 0.0]: 207.0 => [152.0]
13765796.0 [CAN 0.0]: 240.0 => [152.0]
13765799.0 [CAN 0.0]: 786432.0 => [152.0]
13765802.0 [CAN 0.0]: 851967.0 => [152.0]
13765805.0 [CAN 0.0]: 983040.0 => [152.0]
Uptime value is: 128.0
13766791.0 [CAN 0.0]: 192.0 => [128.0]
13766794.0 [CAN 0.0]: 207.0 => [128.0]
13766797.0 [CAN 0.0]: 240.0 => [128.0]
13766800.0 [CAN 0.0]: 786432.0 => [128.0]
13766803.0 [CAN 0.0]: 851967.0 => [128.0]
13766805.0 [CAN 0.0]: 983040.0 => [128.0]
```

It would certainly appear that filtering is not functional unless I have done it wrong.  @brentpicasso does that filter value and mask work in your book?

**Comment by stieg** [comment]
To be thorough I set the changed the following parameters as well

```lua
CAN_REG_ADDR_FAIL_1 = 0x10
CAN_EXT_ADDR_FAIL_1 = 0x10000
```

Still same failure

**Comment by stieg** [comment]
Fixed in 2.10.0
